### ISSUE TITLE
Connecting two Thingsboard instances to the same database for migration period

### ISSUE THREAD

**Issue by dmohns** [thingsboard/thingsboard#10883]
## Component

 * Installation

## Description

**TL;DR:** Is it possible to connect two Thingsboard instances (same version, same configuration, etc..) to the same database? 

**Long Version:** Being aware of the X/Y problem let me explain below, what I want to achieve.

- In our setup we are running Thingsboard `3.6.3` on GKE in monolith setup
- All Kubernetes components sit in a same namespace, let's say `foobar`
- We need to rename the namespace to, let's say `thingsboard` 
- It seems to not be possible to rename Kubernetes namespaces (see https://www.github.com/kubernetes/kubernetes/issues/43867) in place
- Preferably, this reduces user-facing downtime to a minimum

Given the above observations out approach would be to

1. Create the new namespace `thingsboard` and spin up the entire Thingsboard infrastructure pointing to the existing database
2. The `thingsboard` namespace will use a different DNS and IP address which is not a problem
3. Users, can now access Thingsboard via the old and the new namespace which gives them time to migrate
4. Decommission the old `foobar` namespace

My question is: Will point 3. work? Or is this not possible due to some Thingsboard internals?

## Environment

 * OS:  https://thingsboard.io/docs/user-guide/install/pe/cluster/gcp-monolith-setup/
 * ThingsBoard: `3.6.3`
 * Browser: Chrome (Version 125.0.6422.77 (Official Build) (arm64))

## Disclaimer

We appreciate your contribution whether it is a bug report, feature request, or pull request with improvement (hopefully). Please comply with the [Community ethics policy](https://docs.github.com/en/site-policy/acceptable-use-policies/github-acceptable-use-policies), and do not expect us to answer your requests immediately. Also, do not treat *GitHub issues* as a support channel.

**Comment by dmohns** [comment]
Thanks for your response!

> You need to ensure those "different" ThingsBoard applications are aware of each other (zookeeper) and have same cache (redis). They should also have one common load balancer setup.

I'm using Thingsboard in monolith setup. So, I will have separate cache and loadbalancing if I'm not mistaken. However, the two instances will run on entirely different DNS/URLs.

> Otherwise, there will be issues, such as message processing duplication and wrong API responses.

Does this still apply to my case? I would have hoped that (because the instances are isolated from another) that message processing is consistent in itself. But I'm not at all familiar with the internals of message processing 🙈

**Comment by trikimiki** [comment]
@dmohns 

while it is hard to predict exactly what's gonna happen, rest assured two TBs that have no clue about each other will cause issues

> I would have hoped that (because the instances are isolated from another) that message processing is consistent in itself.

this is more about "system" messaging, like device state, API consistency, etc.
basically, every action that saves something to the DB might have unexpected results.

**Comment by dmohns** [comment]
> @dmohns
> 
> while it is hard to predict exactly what's gonna happen, rest assured two TBs that have no clue about each other will cause issues
> 
> > I would have hoped that (because the instances are isolated from another) that message processing is consistent in itself.
> 
> this is more about "system" messaging, like device state, API consistency, etc. basically, every action that saves something to the DB (or cache) might have unexpected results.

Alright. Makes sense. Thanks for the response. 

As for my use-case, I will go for a scheduled maintenance downtime and will put the instance offline for the time of namespace migration 👍
